homeassistant/components/sonos/number.py

```diff
@@ -142,3 +173,369 @@ def native_value(self) -> float:
+    """
+
+    _attr_translation_key = "group_volume"
+    _attr_icon = "mdi:volume-high"
+    _attr_native_min_value = 0
+    _attr_native_max_value = 100
+    _attr_native_step = 1
+    _attr_mode = NumberMode.SLIDER
+
+    def __init__(self, speaker: SonosSpeaker, config_entry: SonosConfigEntry) -> None:
+        """Initialize the Sonos group volume number entity."""
+        super().__init__(speaker, config_entry)
+        self._attr_unique_id = f"{self.soco.uid}-group_volume"
+
+        self._coord_uid: str | None = None
+        self._group_uid: str | None = None
+
+        self._unsubscribe_coord: Callable[[], None] | None = None
+        self._unsubscribe_member: Callable[[], None] | None = None
+        self._unsubscribe_activity: Callable[[], None] | None = None
+        self._unsubscribe_gv_signal: Callable[[], None] | None = None
+        self._unsubscribe_gv_req: Callable[[], None] | None = None
+        self._delay_unsubscribe: Callable[[], None] | None = None
+
+        self._value: int | None = None
+        self._last_rebind_time: float = 0.0
+        self._bootstrap: bool = True  # startup settling window
+
+    def _coordinator_soco(self) -> SoCo:
+        """Return the coordinator SoCo for this speaker."""
+        return (self.speaker.coordinator or self.speaker).soco
+
+    def _current_group_uid(self) -> str | None:
+        group = getattr(self._coordinator_soco(), "group", None)
+        return getattr(group, "uid", None)
+
+    def _is_grouped(self) -> bool:
+        """Return True if this speaker is currently grouped with others."""
+        if (group := getattr(self._coordinator_soco(), "group", None)) and (
+            members := getattr(group, "members", None)
+        ):
+            return len(members) > 1
+        return False
+
+    def _is_coordinator(self) -> bool:
+        return (self.speaker.coordinator or self.speaker).uid == self.speaker.uid
+
+    def _schedule_delayed_refresh(self, seconds: float = GV_REFRESH_DELAY) -> None:
+        """Schedule a short delayed refresh on the HA loop (thread-safe)."""
+
+        def _schedule() -> None:
+            # Cancel any pending timer
+            if self._delay_unsubscribe is not None:
+                self._delay_unsubscribe()
+                self._delay_unsubscribe = None
+
+            loop = self.hass.loop
+            if not loop.is_running() or loop.is_closed():
+                return
+
+            async def _delayed_refresh(_now: datetime) -> None:
+                self._delay_unsubscribe = None
+                self._rebind_for_topology_change()
+
+                # During bootstrap, let members populate from the coordinator once
+                if (
+                    self._bootstrap
+                    and self._is_grouped()
+                    and not self._is_coordinator()
+                    and self._value is None
+                ):
+                    await self._async_initial_populate()
```

Comment:
The tests do not cover this section of the code. HA does not execute line 254 when running in debug mode; starting up with speakers that are grouped or un-grouped.